### ISSUE TITLE
Connecting to a PowershellSDK-Hosted PowershellProcess with Enter-PSHostProcess fails

### ISSUE THREAD

**Issue by Herrmel** [PowerShell/PowerShell#9317]
# Steps to reproduce
Exampleprogramm using the Powershell.SDK:
```Csharp
class Program
    {
        static void Main(string[] args)
        {
            PowerShell ps = PowerShell.Create();
            while (true)
            {
                var pipe=ps.Runspace.CreatePipeline();
                pipe.Commands.AddScript(Console.ReadLine());
                pipe.Commands.Add("out-string");
                Console.Write(pipe.Invoke().FirstOrDefault());
            }
        }
    }
```
start and from a default powershellconsole try:
```powershell
Get-PSHostProcessInfo
Enter-PSHostProcess <ProcessId>
```

# Expected behavior

Get-PSHostProcessInfo lists the Process of the application and connecting with the correct ProcessId works. 

# Actual behavior

Get-PSHostProcessInfo lists the Process of the application but on entering the following error occurrs:
```none
Enter-PSHostProcess : The term 'Measure-Object' is not recognized as the name of a cmdlet, function, script file, or operable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ Enter-PSHostProcess 10688
+ ~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidOperation: (Microsoft.Power...tProcessCommand:EnterPSHostProcessCommand) [Enter-PSHostProcess], RemoteException
+ FullyQualifiedErrorId : EnterPSHostProcessCannotPushRunspace,Microsoft.PowerShell.Commands.EnterPSHostProcessCommand
```

# Environment data

```none
Name                           Value
----                           -----
PSVersion                      6.2.0
PSEdition                      Core
GitCommitId                    6.2.0
OS                             Microsoft Windows 10.0.17134 
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

If there is any better way for remoteaccessing a selfhosted powershell application please tell me as well. I didnt really find anything related to this on google. The Docs for the PowershellSDK are not mentioning this aswell or I couldn't find it. Currently the plan is to remoteconnect (with Enter-PSSession) to the PC running my application and from there use Enter-PSHostProcess.
Thank you

**Comment by Herrmel** [comment]
Is there anything i left unclear or are you unable to reproduce? Did I understand the functionality of Enter-PSHostProcess wrong? Any answer would be appreciated

**Comment by Herrmel** [comment]
Btw both sides have the Measure-Object command when i search them per Get-Command...
Can no one here help me?

I'll explain my plan a little further maybe that helps: My programm uses the powershell.SDK for ConsoleInputs. Since Powershell already has remotecapabilities I would like to use these for a remote console. The problem is I need to connect to the process of my Programm to get the ability to communicate with the already running programm per static function calls. So I've searched a lot and [here](https://stackoverflow.com/questions/39376165/getting-enter-pshostprocess-behavior-via-pssessionconfiguration-file) they mention a double-Hop to connect to the process remotly. The problem is I cant even connect to my programm running powershell with Enter-PSHostProcess because I get the Error above.

**Comment by iSazonov** [comment]
We should implement interactive host to use Enter-PSHostProcess

**Comment by martinweiss92** [comment]
Hi @Herrmel. 
I also encountered this problem after upgrading a Windows Powershell hosting project to Powershell 7.4.3.

I was able to fix the error message described above by copying the modules contained in the Microsoft.PowerShell.SDK nuget package, directly to my output directory: => `.\runtimes\win\lib\net8.0\Modules\` in parallel to the system.management.automation.dll.

This way the required 'Microsoft.PowerShell.Utility' module can be successfully be imported, as it is searched in $PSHome\Modules.

unfortunately, the enter-pshostprocess cmdlet then still failed with the following error and the unhelpful error message:
`Enter-PSHostProcess : Unable to connect to application domain name of process . Error: .`

from the failed remote-runspace i was able to identify this as the cause: 
`System.Management.Automation.Remoting.PSRemotingDataStructureException: Remote host method data decoding is not supported for type System.Management.Automation.Host.Size`.

as mentioned by @iSazonov, this requires an interactive host.
in my case I am not dealing with a console application but I can successfully create my runspace with a custom PSHost implementation.
microsoft provides a great example in which the [IHostSupportsInteractiveSession](https://learn.microsoft.com/en-us/powershell/scripting/developer/hosting/host05-sample?view=powershell-7.4#example-2) interface is implemented.

now enter-pshostprocess works for me without any problems.
